Userland/Applications/FileManager/PropertiesWindow.cpp

```diff
@@ -251,6 +257,98 @@ ErrorOr<void> PropertiesWindow::create_audio_tab(GUI::TabWidget& tab_widget, Non
     return {};
 }
 
+ErrorOr<void> PropertiesWindow::create_image_tab(GUI::TabWidget& tab_widget, NonnullRefPtr<Core::MappedFile> mapped_file, StringView mime_type)
+{
+    auto image_decoder = Gfx::ImageDecoder::try_create_for_raw_bytes(mapped_file->bytes(), mime_type);
+    if (!image_decoder)
+        return {};
+
+    auto tab = TRY(tab_widget.try_add_tab<GUI::Widget>("Image"_short_string));
+    TRY(tab->load_from_gml(properties_window_image_tab_gml));
+
+    // FIXME: The JPEG decoder doesn't decode its header (and so, know what its size is) until we request either a frame,
+    //        or the ICC data. We don't want a frame, but we do want the ICC data later, so we do this right at the start.
+    //        This is a hack but it should be fine. :^)
+    auto embedded_icc_bytes = TRY(image_decoder->icc_data());
+
+    tab->find_descendant_of_type_named<GUI::Label>("image_type")->set_text(TRY(String::from_utf8(mime_type)));
+    tab->find_descendant_of_type_named<GUI::Label>("image_size")->set_text(TRY(String::formatted("{} x {}", image_decoder->width(), image_decoder->height())));
+
+    String animation_text;
+    if (image_decoder->is_animated()) {
+        auto loops = image_decoder->loop_count();
+        auto frames = image_decoder->frame_count();
+        StringBuilder builder;
+        if (loops == 0) {
+            TRY(builder.try_append("Loop indefinitely"sv));
+        } else if (loops == 1) {
+            TRY(builder.try_append("Once"sv));
+        } else {
+            TRY(builder.try_appendff("Loop {} times"sv, loops));
+        }
+        TRY(builder.try_appendff(" ({} frames)"sv, frames));
+
+        animation_text = TRY(builder.to_string());
+    } else {
+        animation_text = "None"_short_string;
+    }
+    tab->find_descendant_of_type_named<GUI::Label>("image_animation")->set_text(move(animation_text));
+
+    auto hide_icc_group = [&tab](String profile_text) {
+        tab->find_descendant_of_type_named<GUI::Label>("image_has_icc_profile")->set_text(profile_text);
+        tab->find_descendant_of_type_named<GUI::Widget>("image_icc_group")->set_visible(false);
+    };
+
+    if (embedded_icc_bytes.has_value()) {
+        auto icc_profile_or_error = Gfx::ICC::Profile::try_load_from_externally_owned_memory(embedded_icc_bytes.value());
+        if (icc_profile_or_error.is_error()) {
+            hide_icc_group(TRY("Present but invalid"_string));
+        } else {
+            auto icc_profile = icc_profile_or_error.release_value();
+            auto extract_string = [](Gfx::ICC::Profile& profile, Gfx::ICC::TagSignature signature) -> ErrorOr<String> {
```

Comment:
This should probably move to LibGfx/ICC/SomeFile.h at some point, seems generally useful (…but we can move it once we need it in a 2nd place, having it here works for me for now).